### PR TITLE
 gnrc_sock: provide asynchronous event implementation

### PR DESCRIPTION
**Pull request by miri64**
This provides an implementation for GNRC for the asynchronous event management for `sock` as provided in #8149.

### CODE REVIEW

File: sys/net/gnrc/sock/gnrc_sock.c [hunk 1]
```diff
@@ -43,21 +46,51 @@ static void _callback_put(void *arg)
...
+#ifdef SOCK_HAS_ASYNC
+void _netapi_cb(uint16_t cmd, gnrc_pktsnip_t *pkt, void *ctx)
+{
+    if (cmd == GNRC_NETAPI_MSG_TYPE_RCV) {
+        msg_t msg = { .type = GNRC_NETAPI_MSG_TYPE_RCV,
+                      .content = { .ptr = pkt } };
+        gnrc_sock_reg_t *reg = ctx;
+
+        if (mbox_try_put(&reg->mbox, &msg) < 1) {
+            DEBUG("gnrc_sock: dropped message to %p (was full)\n", mbox);
```
Comment (rfuentess) [inline sys/net/gnrc/sock/gnrc_sock.c]: Probably this line should be:

```DEBUG("gnrc_sock: dropped message to %p (was full)\n", (void *)&reg->mbox);```

Comment (miri64) [reply sys/net/gnrc/sock/gnrc_sock.c]: Yepp.

File: sys/net/gnrc/sock/include/sock_types.h [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+#include "event.h"
```
Comment (rfuentess) [inline sys/net/gnrc/sock/include/sock_types.h]: This probably should be: 

```c
#ifdef SOCK_HAS_ASYNC
#include "event.h"
#endif
```

Comment (miri64) [reply sys/net/gnrc/sock/include/sock_types.h]: Doesn't hurt, though it is not necessary

**Comment by miri64** [comment]
Yepp, can be closed for now, as there is a new iteration of asynchronous sock: https://github.com/RIOT-OS/RIOT/pull/11723

**Pull request closed without merging**